### ISSUE TITLE
Middle-ground for wrapping function/macro arguments

### ISSUE THREAD

**Issue by ssokolow** [rust-lang-nursery/rustfmt#1241]

**Comment by nrc** [comment]
cc https://github.com/rust-lang-nursery/fmt-rfcs/issues/86

**Comment by jieyouxu** [comment]
Triage: the special-casing for `assert_eq!` and `assert!` was implemented in #2306, and now formats like:

```
$ rustfmt +stable foo.rs
$ cat foo.rs
assert!(
    1 == 1,
    "12345678901234567890123456789012345678901234567890123456789012345678{}",
    901
);
```

`rustfmt 1.8.0-stable (1159e78c47 2025-09-14)`

**Comment by ssokolow** [comment]
That's literally the "not good" formatting that prompted me to open this in the first place.

I'm asking for something more like this so I can stop using stuff like `#[rustfmt::skip]`:

```
assert!(1 == 1,
        "12345678901234567890123456789012345678901234567890123456789012345678{}", 901);
```

**Comment by jieyouxu** [comment]
Fair enough, reopening.

**Comment by Manishearth** [comment]
I'm not really convinced this warrants a config option. It's not the best styling, but also most config options are a lot more general.

**Comment by ssokolow** [comment]
> I'm not really convinced this warrants a config option. It's not the best styling, but also most config options are a lot more general.

True... but most config options don't deal with stuff that gets repeated in test suites enough that I sometimes `#[rustfmt:skip]` entire `mod test`s.

I'd be OK with such an option being perma-unstable, as long as it exists to be set as part of setting `use_small_heuristics = "Max"`.
